orchestrator/forms/validators.py

```diff
@@ -87,37 +80,43 @@ def __str__(self) -> str:
         return f"value is not a valid enumeration member; permitted: {permitted}"
 
 
-class ProductId(UUID):
-    products: Optional[List[UUID]] = None
+def _validate_is_product(id_: UUID) -> UUID:
+    if not get_product_by_id(id_, join_fixed_inputs=False):
+        raise ValueError("Product not found")
+
+    return id_
+
+
+def _validate_in_products(product_ids: set[UUID], id_: UUID) -> UUID:
+    if product_ids and id_ not in product_ids:
+        raise ProductIdError(enum_values=list(map(str, product_ids)))
+
+    return id_
 
-    @classmethod
-    def __modify_schema__(cls, field_schema: Dict[str, Any]) -> None:
-        kwargs = {"uniforms": {"productIds": cls.products}} if cls.products else {}
-        field_schema.update(format="productId", **kwargs)
 
-    @classmethod
-    def __get_validators__(cls) -> Generator:
-        yield uuid_validator
-        yield cls.is_product
-        if cls.products:
-            yield cls.in_products
+ProductId = Annotated[UUID, AfterValidator(_validate_is_product), Field(json_schema_extra={"format": "productId"})]
 
-    @classmethod
-    def is_product(cls, v: UUID) -> UUID:
-        product = products.get_product_by_id(v)
-        if product is None:
-            raise ValueError("Product not found")
 
-        return v
+def product_id(product_ids: Optional[List[UUID]] = None) -> Type[ProductId]:
+    schema = {"uniforms": {"productIds": product_ids}} if product_ids else {}
+    return Annotated[  # type: ignore[return-value]
+        ProductId,
+        AfterValidator(partial(_validate_in_products, set(product_ids or []))),
+        Field(json_schema_extra={"format": "productId"} | schema),
+    ]
 
-    @classmethod
-    def in_products(cls, v: UUID) -> UUID:
-        if cls.products and v not in cls.products:
-            raise ProductIdError(enum_values=list(map(str, cls.products)))
 
-        return v
+def remove_empty_items(v: list) -> list:
```

Comment:
This was moved to pydantic-forms but it's no longer used there, so I moved it back here. (someone could be using it)

Reply:
I think its still fine to keep in the pydantic-forms as handy function. would need some docs to find in there

Reply:
I don't know what we need this function for and would rather deprecate it :)